src/book-details/BookDetails.tsx

```diff
@@ -0,0 +1,278 @@
+import './BookDetails.css';
+import { useState, useEffect } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { API_ENDPOINTS, API_CONFIG } from '../constants/api';
+import Spinner from '../spinner/Spinner';
+import { BookDetailsAPI } from '../types/book';
+import { BookDetailsProps } from '../types/components';
+
+function BookDetails({ results }: BookDetailsProps) {
+  const { detailsId, page = '1' } = useParams();
+  const navigate = useNavigate();
+
+  const [bookDetailsAPI, setBookDetailsAPI] = useState<BookDetailsAPI | null>(
+    null
+  );
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const bookFromList = results.find((book) => {
+    const cleanKey = book.key?.replace('/works/', '');
+    return cleanKey === detailsId;
+  });
+
+  useEffect(() => {
+    if (!detailsId) return;
+
+    const fetchBookDetails = async () => {
+      setLoading(true);
+      setError(null);
+
+      const startTime = Date.now();
+
+      try {
+        const response = await fetch(
+          `${API_ENDPOINTS.BOOK_DETAILS}/${detailsId}.json`
+        );
+
+        if (!response.ok) {
+          throw new Error(`Failed to fetch book details: ${response.status}`);
+        }
+
+        const data: BookDetailsAPI = await response.json();
+
+        const elapsedTime = Date.now() - startTime;
+        const minLoadingTime = 1500;
+
+        if (elapsedTime < minLoadingTime) {
+          await new Promise((resolve) =>
+            setTimeout(resolve, minLoadingTime - elapsedTime)
+          );
+        }
+
+        setBookDetailsAPI(data);
+      } catch (err) {
+        setError((err as Error).message);
+        console.error('Error fetching book details:', err);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchBookDetails();
+  }, [detailsId]);
+
+  const handleClose = () => {
+    navigate(`/${page}`);
+  };
+
+  const getCoverUrl = (coverId: number): string => {
+    return `${API_CONFIG.COVER_BASE_URL}/${coverId}-L.jpg`;
+  };
```

Comment:
```suggestion
  const getCoverUrl = (coverId: number) => (
  `${API_CONFIG.COVER_BASE_URL}/${coverId}-L.jpg`
  )
```
TS сам тип выведет, указание возвращаемого типа тут избыточно